src/lightning/fabric/cli.py

```diff
@@ -44,7 +47,31 @@ def _get_supported_strategies() -> List[str]:
 if _CLICK_AVAILABLE:
     import click
 
-    @click.command(
+    def _legacy_main() -> None:
+        """Legacy CLI handler for fabric.
+
+        Raises deprecation warning and runs through fabric cli if necessary, else runs the entrypoint directly
+
+        """
+        print("`lightning run model` is deprecated and will be removed in future versions."
+            " Please call `fabric run model` instead.")
+        args = sys.argv[1:]
+        if args and args[0] == "run" and args[1] == "model":
+            _main()
```

Comment:
I think the return would still be necessary, otherwise it could execute the case below